base_layer/wallet/src/base_node_service/handle.rs

```diff
@@ -74,6 +74,14 @@ impl BaseNodeServiceHandle {
         self.event_stream_sender.subscribe().fuse()
     }
 
+    pub async fn get_connected_base_node_state(&mut self) -> Result<ChainMetadata, BaseNodeServiceError> {
+        match self.handle.call(BaseNodeServiceRequest::GetChainMetadata).await?? {
+            BaseNodeServiceResponse::ChainMetadata(Some(v)) => Ok(v),
+            BaseNodeServiceResponse::ChainMetadata(None) => Err(BaseNodeServiceError::NoChainMetadata),
```

Comment:
Don't see why you need a whole new error type for this, just pass the Option out?

Reply:
I see it as more complete. For this to return an error. The wallet must have never talked to a basenode in its entire life or at least be conencted to one.

Reply:
So this is a style nit but I see the fact that the value returned was a None communicates that information which is not actually an Error. An error means something when wrong with the call i.e. getting the connected base node state broke somehow like the db was broken. Getting that state was done without error and the state was None meaning we haven't spoken to a base node.

Reply:
Surprising how much this error that was never supposed to happen came up

Reply:
This points me to there is some other problem in code. Either by the way we handle linking of components or by us not properly linking them up.